apps/rush-lib/src/api/LastInstallFlag.ts

```diff
@@ -26,10 +27,35 @@ export class LastInstallFlag {
    * @param state - optional, the state that should be managed or compared
    */
   public constructor(folderPath: string, state: JsonObject = {}) {
-    this._path = path.join(folderPath, LAST_INSTALL_FLAG_FILE_NAME);
+    this._path = path.join(folderPath, this.flagName);
     this._state = state;
   }
 
+  /**
+   * Gets the current state of the Rush repo. This state is used to compare against
+   * the last-known-good state tracked by the LastInstall flag.
+   * @param rushConfiguration - the configuration of the Rush repo to get the install
+   * state from
+   *
+   * @internal
+   */
+  public static getCommonTempFlag(rushConfiguration: RushConfiguration): LastInstallFlag {
```

Comment:
The `LastInstallFlag` was designed as a conceptually complete API, that allows you to create installation flags with arbitrary bags of JSON state, and which in theory could be reusable by other projects.  It is not meant to be subclassed or to have business logic mixed into it.

I would suggest to centralize these functions in a helper, something like this:
```
export class InstallFlags {
  // common/temp/last-install.flag
  public static getLastInstallFlag(rushConfiguration: RushConfiguration): LastInstallFlag {
    . . .
  }
  // common/temp/last-link.flag
  public static getLastLinkFlag(rushConfiguration: RushConfiguration): LastInstallFlag {
    . . .
  }
}
```

Reply:
While I agree, I plan on removing this flag once we fully deprecate `rush link` and `rush unlink` so this file is temporary, this was done simply to keep the code short and sweet. Personally, I think that the flag file concept should move into `node-core-library` (as we talked about offline). I'd rather change this flag to use a helper when we generalize the concept.

However, if you feel strongly about it, it can be changed